pkg/indexer/rpc_streamer/rpc_log_streamer.go

```diff
@@ -244,12 +258,44 @@ backfillLoop:
 			logger.Error("Context cancelled, stopping watcher")
 			return
 
+		case err, open := <-sub.Err():
+			if !open {
+				logger.Error("subscription channel closed, closing watcher")
+				return
+			}
+
+			logger.Error("subscription error, rebuilding", zap.Error(err))
+			sub, err = r.buildSubscriptionWithBackoff(cfg, innerSubCh)
+			if err != nil {
+				logger.Error("failed to rebuild subscription, closing", zap.Error(err))
+				return
+			}
+
+			logs, err := r.backfillPage(r.ctx, cfg, backfillFromBlockNumber)
+			if err != nil {
+				logger.Error("failed to backfill page, closing", zap.Error(err))
+				return
```

Comment:
if you terminate the goroutine here, what will restart the watcher? Same for all returns in this error block